obspy/core/tests/test_trace.py

```diff
@@ -2721,3 +2721,14 @@ def test_resample_short_traces(self):
 
         assert tr.stats.sampling_rate == 30
         assert tr.data.shape[0] == 1
+
+    def test_long_processing_list(self):
+        """
+        issue 2882
+        """
+        tr = read()[0]
+        with pytest.warns(UserWarning, match='.*maximal length') as record:
+            for _ in range(110):
+                tr.trim(0.01)
```

Comment:
you could set `tr.stats.processing = [''] * 99` and only do the one call emitting the wanring in the context manager

Reply:
Good idea, both implemented.